src/ApiService/ApiService/onefuzzlib/Extension.cs

```diff
@@ -239,7 +239,8 @@ private sealed class Settings {
         var sep = pool.Os == Os.Windows ? "\r\n" : "\n";
 
         if (pool.Os == Os.Windows && scaleSet.Auth is not null) {
-            var sshKey = scaleSet.Auth.PublicKey.Trim();
+            var auth = await _context.SecretsOperations.GetSecretValue<Authentication>(scaleSet.Auth);
+            var sshKey = auth?.PublicKey.Trim();
```

Comment:
Thoughts on using `!` instead of `?` here? I'm thinking we'd probably want to faily early and loud if we're expecting auth to be present for the scaleset but we don't find it.

Otherwise, it'll happily serialize to "null" and be a lot harder to debug

Reply:
added a null check